src/components/VoucherModal/index.tsx

```diff
@@ -1,25 +1,42 @@
 import React, { useState, useEffect, useLayoutEffect } from "react"
 import { Skeleton, SkeletonText } from "@chakra-ui/react"
 
+import {
+  redeemVoucher
+} from "api/voucher"
+
 import usePrevious from "hooks/usePrevious"
 import { dateToString } from "utils/date"
 
-import { Modal, ModalProps, Heading, Text } from "@commitUI/index"
+import { Modal, ModalProps, Heading, Text, Button } from "@commitUI/index"
 
 import styles from "./VoucherModal.module.css"
 
 type Props = Omit<ModalProps, "children"> & {
   voucher?: Voucher
   isValidating: boolean
+  user?: User
+  redeemed?: String[] | undefined
+  onCloseHandler: VoidFunction
 }
 
-const VoucherModal = ({ voucher, isOpen, onClose, isValidating }: Props) => {
-  const currVoucher = voucher?.id
+const VoucherModal = ({ onCloseHandler, redeemed, user, voucher, isOpen, onClose, isValidating }: Props) => {
+  const currVoucher = voucher?.uuid
   const prevVoucher = usePrevious(currVoucher)
   const [loading, setLoading] = useState(
     isValidating || prevVoucher !== currVoucher
   )
 
+  const handleRedeem = async () => {
+    const data = {
+      voucher: voucher?.uuid,
+      email: user?.username + "@u.nus.edu"
```

Comment:
A good practice but optional to change: I usually like putting constants such as this (`@u.nus.edu`) inside its own `.ts` file inside the `/constants` directory. If you'd like you can create a new file on that directory and export the email suffix as a constant.

Rule of thumb: If it is likely that the constant will be used in multiple places throughout the project, it is best to put in in a `constants` file